buildSrc/src/main/groovy/airbyte-java-connector.gradle

```diff
@@ -0,0 +1,90 @@
+/*
+Gradle plugin for Java-based Airbyte connectors.
+Also facilitates importing and working with the Java CDK.
+*/
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.testing.Test
+
+class AirbyteJavaConnectorExtension {
+
+    boolean useLocalCdk = false
+    String cdkVersionRequired
+    List<String> features = []  // e.g. 'db-sources', 'db-destinations'
+    Project project
+
+    AirbyteJavaConnectorExtension(Project project) {
+        this.project = project
+    }
+
+    void addCdkDependencies() {
```

Comment:
Why do we force every user of the `AirbyteJavaConnectorPlugin` to also call `addCdkDependencies()`?  

Is there a reason why this code exist couldn't exist within the `AirbyteJavaConnectorPlugin` instead of this extension and be called from the `apply` method of the plugin? 

Changing the usage from
```
airbyteJavaConnector {
    cdkVersionRequired = '0.1.0'
    features = ['db-sources']
    useLocalCdk = true
}

airbyteJavaConnector.addCdkDependencies()
```
to
```
airbyteJavaConnector {
    cdkVersionRequired = '0.1.0'
    features = ['db-sources']
    useLocalCdk = true
}
```

Reply:
Good call-out. This is something I expect we'll get to soon, if not in PR. I've mostly been tracking down other breaking/blocking dependency issues but I plan to revisit when time allows.

@postamar called this out at well.

Reply:
I tried once more but we're getting weird errors regarding what Gradle will permit. I've added a code comment inline within the plugin file, with action item on what should be debugged in order to enable this.